src/wallet/components/WalletDropdownFundLink.tsx

```diff
@@ -0,0 +1,76 @@
+import { isValidElement, useMemo } from 'react';
+import { FundWalletSvg } from '../../internal/svg/fundWallet';
+import { cn, pressable, text as themeText } from '../../styles/theme';
+import { version } from '../../version';
+import type { WalletDropdownFundLinkReact } from '../types';
+import { getWindowDimensions } from '../utils/getWindowDimensions';
+
+export function WalletDropdownFundLink({
+  className,
+  icon,
+  rel,
+  popupFeatures,
+  openIn = 'tab',
+  target,
+  text = 'Fund wallet',
+  popupSize = 'md',
+}: WalletDropdownFundLinkReact) {
+  const iconSvg = useMemo(() => {
+    if (icon === undefined) {
+      return FundWalletSvg;
+    }
+    if (isValidElement(icon)) {
+      return icon;
+    }
+  }, [icon]);
+
+  const currentURL = window.location.href;
```

Comment:
Something to reflect here, is the cost of creating this variable every time the React component render. Which is usually a lot.

So this is possibly something that should be wrapped by a useEffect.

Also in general for @0xAlec @cpcramer @abcrane123 please watch a few times this video https://www.youtube.com/watch?v=NGxVLnJKhP8 it really helps understand React internals.